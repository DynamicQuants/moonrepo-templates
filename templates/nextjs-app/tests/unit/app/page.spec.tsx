import { render } from '@testing-library/react';

import Home from '../../../src/app/page';

describe('Home page', () => {
  let container: HTMLElement;
  let unmount: () => void;

  beforeAll(async () => {
    const { container: _container, unmount: _unmount } = render(await Home());
    container = _container;
    unmount = _unmount;
  });

  afterAll(() => {
    unmount();
  });

  it('should match the snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('should have the correct tree parent', () => {
    expect(container).toBeInstanceOf(HTMLDivElement);
  });
});
