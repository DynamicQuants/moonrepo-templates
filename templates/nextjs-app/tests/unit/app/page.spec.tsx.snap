// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should match the snapshot 1`] = `
<div>
  <div
    class="font-sans h-screen flex flex-col items-center justify-center"
  >
    <img
      alt="Next.js Logo"
      class="mb-4"
      data-nimg="1"
      decoding="async"
      height="150"
      loading="lazy"
      src="/_next/image?url=%2Fnextjs-logo.webp&w=384&q=75"
      srcset="/_next/image?url=%2Fnextjs-logo.webp&w=256&q=75 1x, /_next/image?url=%2Fnextjs-logo.webp&w=384&q=75 2x"
      style="color: transparent;"
      width="150"
    />
    <h1
      class="text-6xl font-bold text-center"
    >
      Hello World!
    </h1>
    <p
      class="font-mono text-lg text-center mt-4"
    >
      Welcome to your new Next.js app.
    </p>
  </div>
</div>
`;
